api/spec/scanner-adapter-openapi-v1.2.yaml

```diff
@@ -0,0 +1,546 @@
+      description: |
+        A set of custom properties that can further describe capabilities of a given scanner.
+    ScannerCapability:
+      description: |
+        Capability consists of the set of recognized artifact MIME types and the set of scanner report MIME types.
+
+        For example, a scanner capable of analyzing Docker images and producing a vulnerabilities report recognizable
+        by Harbor web console might be represented with the following capability:
+        - consumes MIME types:
+          - `application/vnd.oci.image.manifest.v1+json`
+          - `application/vnd.docker.distribution.manifest.v2+json`
+        - produces MIME types:
+          - `application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0`
+
+        For example, a scanner capable of analyzing artifacts and producing a sbom report recognizable
+        by Harbor might be represented with the following capability:
+        - type: sbom
+        - consumes MIME types:
+          - `application/vnd.oci.image.manifest.v1+json`
+          - `application/vnd.docker.distribution.manifest.v2+json`
+        - produces MIME types:
+          - `application/vnd.security.sbom.report+json; version=1.0`
+      required:
+        - consumes_mime_types
+        - produces_mime_types
+      type: object
+      properties:
+        type:
+          type: string
+          enum:
+          - vulnerability
+          - sbom
+          description: |
+            The type of the capability, for example, 'vulnerability' represents analyzing the artifact then producing the vulnerabilities report,
+            'sbom' represents generating the corresponding sbom for the artifact which be scanned. In order to the backward and forward compatible,
+            the field is optional, we think it's a original 'vulnerability' scan if no such field.
+        consumes_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of the artifacts supported by the scanner to produce the reports specified in the "produces_mime_types". A given
+            mime type should only be present in one capability item.
+          example:
+            - "application/vnd.oci.image.manifest.v1+json"
+            - "application/vnd.docker.distribution.manifest.v2+json"
+        produces_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of reports generated by the scanner for the consumes_mime_types of the same capability record.
+          example:
+            - "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
+    ScanRequest:
+      required:
+        - registry
+        - artifact
+      type: object
+      properties:
+        registry:
+          $ref: '#/components/schemas/Registry'
+        artifact:
+          $ref: '#/components/schemas/Artifact'
+        enabled_capabilities:
```

Comment:
Question: If both `sbom` and `vulnerability` are enabled, what should the scanner do?

Reply:
You mentioned the key point, this place is also a big optimization of this change, yes, like you said that the default is that all the capabilities of the scanner can be scanned, but imagine that in the future, if there are more and more capabilities, and there are more images on the harbor, so a global scan will take a long time and resource cost, so we introduce more fine-grained control this time, maintain backward compatibility, and the default can be considered to be a full-ability scan, However, when harbor specifies certain capabilities to be used and the types of reports that correspond to those capabilities, the scanner can use only a subset of the capabilities based on this information,but enabled_capabilities may be replaced with selected_capabilities can be easier and clearly to understand?

Reply:
When full capabilities are enabled (or selected), should scanners generate all of them (SBOM and vulnerabilities for now)?

Reply:
yes, if the full capabilities enabled, the scanner should generate all them, and harbor will poll these reports independently by the mime type.

Reply:
Which one do you think is clearer, more appropriate, and better expresses the meaning here, `enabled_capabilities` or `selected_capabilities`?

Reply:
`enabled_capabilities` sounds good to me.